### PR TITLE
Add validation checks for references within triggers

### PR DESCRIPTION
**Pull request by amarthadan**
[AN-617](https://api3dao.atlassian.net/browse/AN-617)

### CODE REVIEW

File: packages/airnode-validator/src/config/config.test.ts [hunk 1]
```diff
@@ -185,3 +185,49 @@ it('fails if a securitySchemeName is enabled and it is of type "apiKey" or "http
...
+  const config: Config = JSON.parse(
+    readFileSync(join(__dirname, '../../test/fixtures/interpolated-config.valid.json')).toString()
+  );
+
+  it(`fails if an OIS referenced in a trigger doesn't exist`, () => {
+    const invalidConfig = {
+      ...config,
+      triggers: {
+        ...config.triggers,
+        rrp: [{ ...config.triggers.rrp[0], oisTitle: 'nonexistingois' }],
```
Comment (Siegrift) [inline packages/airnode-validator/src/config/config.test.ts:199]: Nit:

```suggestion
        rrp: [{ ...config.triggers.rrp[0], oisTitle: 'nonExistingOis' }],
```

File: packages/airnode-validator/src/config/config.ts [hunk 2]
```diff
@@ -223,6 +223,39 @@ const validateTemplateSchemes: SuperRefinement<{
   }
 };
 
+const validateTriggersReferences: SuperRefinement<{
+  ois: OIS[];
+  triggers: Triggers;
+}> = (config, ctx) => {
+  Object.entries(config.triggers).forEach(([triggerName, triggers]) => {
```
Comment (Siegrift) [inline packages/airnode-validator/src/config/config.ts:230]: Nit: prefer lodash

```suggestion
  forEach(config.triggers, (triggers, triggerSection) => {
```

File: packages/airnode-validator/src/config/config.ts [hunk 3]
```diff
@@ -223,6 +223,39 @@ const validateTemplateSchemes: SuperRefinement<{
...
+    triggers.forEach((trigger, index) => {
+      const { oisTitle, endpointName } = trigger;
+
+      // Check that the OIS with the ois title from trigger exists
```
Comment (Siegrift) [inline packages/airnode-validator/src/config/config.ts:234]:
```suggestion
      // Check that the OIS with the "oisTitle" from the trigger exists
```

**Review by Siegrift — APPROVED** [review]
👍 LGTM

**Comment by amarthadan** [comment]
@Siegrift Can you review this again, please? I've removed some checks from `airnode-node` code that are now handled in validator and want to make sure I didn't mess up anything.

**Review by Siegrift — APPROVED** [review]
👍 Good catch removing the not needed code.

**Pull request merged**